internal/core/src/common/Vector.h

```diff
@@ -146,6 +200,19 @@ class ColumnVector final : public SimpleVector {
         return is_bitmap_;
     }
 
+    void
+    resize(vector_size_t new_size, bool setNotNull = true) override {
+        AssertInfo(!is_bitmap_, "Cannot resize bitmap column vector");
+        BaseVector::resize(new_size, setNotNull);
+        ResizeScalarFieldData(type(), new_size, values_);
+        valid_values_.resize(new_size);
+    }
+
+    void
+    append(const ColumnVector& other) {
+        values_->FillFieldData(other.GetRawData(), other.size());
+    }
+
```

Comment:
in the iterative computing framework, result vectors are returned batch by batch, so we need to append one batch result into the final returned result, this method involves memory copy, may need to be optimized